agent/spec/lib/kontena/workers/node_info_worker_spec.rb

```diff
@@ -214,4 +214,22 @@
       expect(subject.private_ip).to eq('192.168.2.10')
     end
   end
+
+  describe '#publish_node_stats' do
+    it 'sends stats via rpc' do
+      expect(rpc_client).to receive(:notification).once do |key, msg|
```

Comment:
You should be able to compose rspec matchers to do this, without having to use a block:

```ruby
expect(rpc_client).to receive(:notification).once.with('/nodes/stats', [Hash])
```

Or even better for this spec:

```ruby
expect(rpc_client).to receive(:notification).once.with('/nodes/stats', [hash_including(time: String)])
```